5_distance_analysis/nbconverted/1_centroid_distance.py

```diff
@@ -0,0 +1,681 @@
+    valid_ids[:5]
+
+
+# In[11]:
+
+
+if min_sample_count is not None:
+    results_loss_df = results_loss_df.loc[valid_ids, :]
+    
+print(results_loss_df.shape)
+results_loss_df.head()
+
+
+# In[12]:
+
+
+class_counts_df = {}
+results_df = {}
+class_names = {}
+counts_columns = None
+results_columns = None
+
+# get distances for copy gain, for class 1/3/4 genes
+for identifier, gain_df in park_gain_info.items():
+    
+    class_name = gain_df.head(1).class_name.values[0]
+    if class_name == 'class 2':
+        continue
+        
+    results = get_centroids_and_distance(identifier, gain_df, 'mean')
+    
+    if counts_columns is None:
+        counts_columns = results[0]
+    else:
+        assert counts_columns == results[0]
+        
+    if results_columns is None:
+        results_columns = ['{}/{}'.format(i, j) for i, j in results[1]]
+            
+    class_names[identifier] = class_name
+    class_counts_df[identifier] = results[2]
+    results_df[identifier] = results[3]
+    
+class_counts_gain_df = (
+    pd.DataFrame(class_counts_df.values(),
+                 index=class_counts_df.keys(),
+                 columns=counts_columns)
+      .merge(pd.Series(class_names).rename('class_name'),
+             left_index=True, right_index=True)
+)
+    
+results_gain_df = (
+    pd.DataFrame(results_df.values(),
+                 index=results_df.keys(),
+                 columns=results_columns)
+      .merge(pd.Series(class_names).rename('class_name'),
+             left_index=True, right_index=True)
+)
+    
+print(class_counts_gain_df.shape)
+class_counts_gain_df.head()
+
+
+# In[13]:
+
+
+if min_sample_count is not None:
+    valid_rows = (class_counts_gain_df
+        .drop(columns=['class_name'])
+        .astype(int) 
+        > min_sample_count
+    ).all(axis=1)
+    print('Valid rows:', valid_rows.sum(), '/', class_counts_gain_df.shape[0])
+    valid_ids = class_counts_gain_df.index[valid_rows]
+    valid_ids[:5]
+
+
+# In[14]:
+
+
+if min_sample_count is not None:
+    results_gain_df = results_gain_df.loc[valid_ids, :]
+    
+print(results_gain_df.shape)
+results_gain_df.head()
+
+
+# ### Plot centroid distance results
+# 
+# To make our plots, we'll just get rid of NaN rows (i.e. genes/cancer types that don't have at least one sample in each "hit" category).
+
+# In[15]:
+
+
+sns.set({'figure.figsize': (18, 12)})
+
+fig, axarr = plt.subplots(2, 2)
+
+# plot class counts distributions for copy loss
+for ix, class_name in enumerate(['class 1', 'class 2', 'class 4']):
+    ax = axarr[ix // 2, ix % 2]
+    # convert dataframe to long-form to plot it
+    plot_df = (class_counts_loss_df[class_counts_loss_df.class_name == class_name]
+      .drop(columns='class_name')
+      .dropna(axis='index')
+      .reset_index()
+      .rename(columns={'index': 'identifier'})
+      .melt(id_vars='identifier', value_name='count', var_name='num_hits')
+    )
+    sns.kdeplot(data=plot_df, x='count', hue='num_hits', ax=ax)
```

Comment:
How hard is it to add in headers to the individual plots? I'm guessing this is class 1-4, but wouldn't hurt to be sure, especially because Class 4 I would expect to have something but it's the blank one?

Reply:
Just saw line 384, oops. This is still an argument for adding titles to each plot. Be sure to idiotproof your code, because idiots (me) will read it.